core/templates/domain/exploration/read-only-exploration-backend-api.service.spec.ts

```diff
@@ -16,241 +16,209 @@
-  }));
-
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+      }
+    },
+    version: 1,
+    state_classifier_mapping: {}
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule]
+    });
+    roebas = TestBed.get(ReadOnlyExplorationBackendApiService);
+    httpTestingController = TestBed.get(HttpTestingController);
   });
 
-  it('should successfully fetch an existing exploration from the backend',
-    function() {
-      const successHandler = jasmine.createSpy('success');
-      const failHandler = jasmine.createSpy('fail');
-
-      $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-        sampleDataResults);
-      ReadOnlyExplorationBackendApiService.fetchExploration(
-        '0', null).then(successHandler, failHandler);
-      $httpBackend.flush();
-
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
-      expect(failHandler).not.toHaveBeenCalled();
-    }
-  );
+  afterEach(() => {
+    httpTestingController.verify();
+  });
 
   it('should successfully fetch an existing exploration with version from' +
-    ' the backend', function() {
+    ' the backend', fakeAsync(() => {
     const successHandler = jasmine.createSpy('success');
     const failHandler = jasmine.createSpy('fail');
 
-    $httpBackend.expect('GET', '/explorehandler/init/0?v=1').respond(
-      sampleDataResults);
-    ReadOnlyExplorationBackendApiService.fetchExploration(
-      '0', 1).then(successHandler, failHandler);
-    $httpBackend.flush();
+    roebas.fetchExploration('0', 1).then(successHandler, failHandler);
+
+    let req = httpTestingController.expectOne(
+      '/explorehandler/init/0?v=1');
+    expect(req.request.method).toEqual('GET');
+    req.flush(sampleDataResults);
+    flushMicrotasks();
 
     expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
     expect(failHandler).not.toHaveBeenCalled();
-  });
-
-  it('should load a cached exploration after fetching it from the backend',
-    function() {
-      const successHandler = jasmine.createSpy('success');
-      const failHandler = jasmine.createSpy('fail');
-
-      // Loading a exploration the first time should fetch it from the backend.
-      $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-        sampleDataResults);
-      ReadOnlyExplorationBackendApiService.loadExploration(
-        '0', null).then(successHandler, failHandler);
-      $httpBackend.flush();
-
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
-      expect(failHandler).not.toHaveBeenCalled();
-
-      // Loading a exploration the second time should not fetch it.
-      ReadOnlyExplorationBackendApiService.loadExploration(
-        '0', null).then(successHandler, failHandler);
-
-      expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
-      expect(failHandler).not.toHaveBeenCalled();
-    }
-  );
```

Comment:
FYI for all reviewers, this test is not valid. The test statement says: "load a cached exploration after fetching it from the backend". In ReadonlyExplorationBackendApiService(https://github.com/oppia/oppia/pull/11060/files#diff-ad5e5d53ed128541fd8a047e67ce9224b1d52165295b26a6b8ed8091fe6e900cL116-L132), the comment above the `loadExploration` function explicitly mentions that the function doesn't use or modify the cache in any way. What's even more baffling is this test passes when it should have failed. There are two HTTP requests made but only one is accounted for. This test should have thrown error in afterEach where we verify for `noOutstandingRequests` but it doesn't.